bench/Main.hs

```diff
@@ -218,14 +218,16 @@ runBenchmarks benchmarks = do
     putStrLn $ intercalate ", " $ map (showDuration . snd) results
```

Comment:
How big a deal would it be to send the benchmark changes in via a different PR? Seems totally unrelated to the rest of it, and one for @pepeiborra to review.

Reply:
Agreed. I've got a number of changes for the benchmark suite myself, so a separate PR that can be reviewed and merged quickly would be much appreciated.

Reply:
I've pushed my benchmarks changes in #625